parsers/rhea/sql.go

```diff
@@ -0,0 +1,182 @@
+package rhea
+
+import (
+	"database/sql"
+	"errors"
+)
+
+/******************************************************************************
+
+Database functions
+
+These functions take in the rhea.rdf.gz dump file and import them into an sqlite
+database. This database can be used locally for mapping functions, but more importantly,
+it is used because it can enforce the relationships between each part of Rhea and alert
+the user if the parser is failing to pick up on any of the relationships in the Rhea
+database.
+
+******************************************************************************/
+
+// RheaSqlite inserts the Rhea database into an SQLite database with proper normalization for advanced queries.
+func RheaSqlite(db *sql.DB, rhea Rhea) error {
+	// Start transaction with database for insertion. This ensures if there are any problems, they are seamlessly rolled back
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	schema := `
+	-- Rhea tables --
+	CREATE TABLE IF NOT EXISTS chebi (
+		accession TEXT PRIMARY KEY,
+		subclassof TEXT REFERENCES chebi(accession)
+	);
+	
+	CREATE TABLE IF NOT EXISTS compound (
+		id INT,
+		accession TEXT PRIMARY KEY,
+		position TEXT,
+		name TEXT,
+		htmlname TEXT,
+		formula TEXT,
+		charge TEXT,
+		chebi TEXT REFERENCES chebi(accession),
+		polymerizationindex TEXT,
+		compoundtype TEXT NOT NULL CHECK(compoundtype IN ('SmallMolecule', 'Polymer', 'GenericPolypeptide', 'GenericPolynucleotide', 'GenericHeteropolysaccharide'))
+	);
+	
+	CREATE TABLE IF NOT EXISTS reactivepart (
+		id INT,
```

Comment:
What is the point of this ID if we are using `accession` as the primary key? Similar Q for other tables.

Reply:
`accession` is the primary key used in Rhea when referring to things (like in the Rhea RDF, when objects refer to each other they use accession, not ID). ID is, however, used for referencing stuff to outside databases, like Uniprot.